readme.md

```diff
@@ -3,71 +3,704 @@
+});
 
-    pkg: grunt.file.readJSON('package.json'),
-      jshint: {
-        all:  ['Gruntfile.js', 'app/**/*.js', '*.js']
-      }
+```
+In your terminal, run `$ gulp`.
 
-  });
+>This will have the library look for a default task in your `gulpfile.js`. It will then execute the callback that you define for your task.
+
+You should see something like the following:
+
+```bash
+[10:37:37] Using gulpfile ~/wdi8/lessons/build-tools/gulp-example/gulpfile.js
+[10:37:37] Starting 'default'...
+I am the default task!
+[10:37:37] Finished 'default' after 104 μs
+```
+
+<details>
+<summary>
+Q. Does the string `'default'` matter what we call it?
+</summary>
+<br>
+
+```
+Yes, and no. While you can certainly change `default` to `pizza`, it would not be very descriptive of the task.
+
+Also, whatever you define 'name' as in your task: gulp.task('name', callbackFunction)
+
+You need to run the following in our command line:
+
+$ gulp <name>
+
+```
+<br>
+<br>
+</details>
+
+
+## STOP
+
+## Review (5 min)
+
+Let's Review and Answer questions!
+
+## Plugins (5 min)
+
+[Gulp plugins](http://gulpjs.com/plugins/)
+
+Gulp has many additional plugins that we can use in our applications. We need to install them each individually using `npm install <dependency-name> --save-dev`
+
+>Note we writing `--save-dev` instead of `--save`.
+>This adds a separate  `"devDependencies"` object in which we will be saving our dependences. Don't worry too much about the difference for now, just know these plugins would be used differently in production!
+
+## You-Do: Jshint Plugin Example (15 min)
+
+We are going to be working with the `jshint` and `jshint-stylish` plugins so we can easily identify any javascript errors in our code!
+
+```bash
+$ npm install jshint gulp-jshint --save-dev
+$ npm install --save-dev jshint-stylish
+```
+
+In our `gulpfile.js`, let's add the following modules for `jshint`:
+
+```js
+var gulp   = require('gulp');
+var jshint = require('gulp-jshint');
+var stylish = require('jshint-stylish');
+```
+>Nothing new here, just requiring our dependencies we will be using
+
+Now, let's add our task!
+
+```js
+gulp.task('jshint', function() {
+  return gulp.src('./js/*.js')
+  .pipe(jshint())
+  .pipe(jshint.reporter(stylish));
+});
+```
+<details>
+<summary>
+Q. What is `gulp.src`?
+</summary>
+<br>
+
+```
+
+gulp.src is specifying our file source paths. It can also take an array of source paths.
+
+
+```
+<br>
+<br>
+</details>
+
+
+<details>
+<summary>
+Q. What do you think `'./js/*.js'` represents ?
+</summary>
+<br>
+
+```
+Linting all of of javascript files in our `js` directory
+
+```
+<br>
+<br>
+</details>
+
+
+<details>
+<summary>
+Q. What is `.pipe()` in gulp?
+</summary>
+<br>
+
+```
+.pipe() is used to pipe the source file(s) into a plugin. These pipes can chain tasks together so you can add as many plugins as you need!
+
+```
+<br>
+<br>
+</details>
+--------------
+
+Now, run the following in your command line:
+
+``` bash
+$ gulp jshint
```

Comment:
is it possible to run this task from the default task so all you have to type is `gulp`?